source/includes/authentication.cpp

```diff
@@ -45,4 +45,14 @@ auto client = mongocxx::client(uri);
 auto uri = mongocxx::uri("mongodb://<db_username>:<db_password>@<hostname>:<port>/?"
                          "authMechanism=PLAIN&tls=true");
 auto client = mongocxx::client(uri);
-// end-plain
\ No newline at end of file
+// end-plain
+
+// start-auth-err
+try {
+    auto uri = mongocxx::uri("mongodb://<db_username>:<db_password>@<hostname>:<port>/?"
+                         "authSource=admin&authMechanism=SCRAM-SHA-256");
+    auto client = mongocxx::client(uri);
+} catch (const mongocxx::exception& ex) {
+    std::cerr << "Error: " << ex.code() << std::endl;
```

Comment:
```suggestion
    std::cerr << "Error: " << ex.what() << std::endl;
```
Unless it is necessary, recommend users to inspect `.what()` (informative message string) rather than `.code()` (raw information about the error code value and category/domain it belongs to). For example, `ex.code()` alone will merely print something like:
```
Error: mongodb:11
```
whereas `ex.what()` will instead print:
```
Error: Authentication failed.: generic server error
```